axum-macros/src/from_request.rs

```diff
@@ -0,0 +1,499 @@
+        return None;
+    }
+
+    let args = match &segment.arguments {
+        syn::PathArguments::AngleBracketed(args) => args,
+        syn::PathArguments::Parenthesized(_) | syn::PathArguments::None => return None,
+    };
+
+    let ty = if args.args.len() == 2 {
+        args.args.first().unwrap()
+    } else {
+        return None;
+    };
+
+    if let syn::GenericArgument::Type(ty) = ty {
+        Some(ty)
+    } else {
+        None
+    }
+}
+
+fn extract_each_field_rejection(
+    ident: &syn::Ident,
+    fields: &syn::Fields,
+    vis: &syn::Visibility,
+) -> syn::Result<TokenStream> {
+    let rejection_ident = rejection_ident(ident);
+
+    let variants = fields
+        .iter()
+        .map(|field| {
+            let FromRequestAttrs { via } = parse_attrs(&field.attrs)?;
+
+            let field_ty = &field.ty;
+            let ty_span = field_ty.span();
+
+            let variant_name = rejection_variant_name(field)?;
+
+            let extractor_ty = if let Some((_, path)) = via {
+                if let Some(inner) = peel_option(field_ty) {
+                    quote_spanned! {ty_span=>
+                        Option<#path<#inner>>
+                    }
+                } else if let Some(inner) = peel_result_ok(field_ty) {
+                    quote_spanned! {ty_span=>
+                        Result<#path<#inner>, TypedHeaderRejection>
+                    }
+                } else {
+                    quote_spanned! {ty_span=> #path<#field_ty> }
+                }
+            } else {
+                quote_spanned! {ty_span=> #field_ty }
+            };
+
+            Ok(quote_spanned! {ty_span=>
+                #[allow(non_camel_case_types)]
+                #variant_name(<#extractor_ty as ::axum::extract::FromRequest<::axum::body::Body>>::Rejection),
+            })
+        })
+        .collect::<syn::Result<Vec<_>>>()?;
+
+    let impl_into_response = {
+        let arms = fields
+            .iter()
+            .map(|field| {
+                let variant_name = rejection_variant_name(field)?;
+                Ok(quote! {
+                    Self::#variant_name(inner) => inner.into_response(),
+                })
+            })
+            .collect::<syn::Result<Vec<_>>>()?;
+
+        quote! {
+            impl ::axum::response::IntoResponse for #rejection_ident {
+                fn into_response(self) -> ::axum::response::Response {
+                    match self {
+                        #(#arms)*
+                    }
+                }
+            }
+        }
+    };
+
+    let impl_display = {
+        let arms = fields
+            .iter()
+            .map(|field| {
+                let variant_name = rejection_variant_name(field)?;
+                Ok(quote! {
+                    Self::#variant_name(inner) => inner.fmt(f),
+                })
+            })
+            .collect::<syn::Result<Vec<_>>>()?;
+
+        quote! {
+            impl ::std::fmt::Display for #rejection_ident {
```

Comment:
There's this + two more `impl ::`s that don't have `#[automatically_derived]` yet.